pyro/contrib/gp/models/gpr.py

```diff
@@ -65,44 +65,38 @@ class GPRegression(GPModel):
         a covariance matrix to help stablize its Cholesky decomposition.
     :param str name: Name of this model.
     """
-    def __init__(self, X, y, kernel, noise=None, mean_function=None, jitter=1e-6,
-                 name="GPR"):
-        super(GPRegression, self).__init__(X, y, kernel, mean_function, jitter, name)
+    def __init__(self, X, y, kernel, noise=None, mean_function=None, jitter=1e-6):
+        super(GPRegression, self).__init__(X, y, kernel, mean_function, jitter)
 
-        noise = self.X.new_ones(()) if noise is None else noise
+        noise = self.X.new_tensor(1.) if noise is None else noise
         self.noise = Parameter(noise)
-        self.set_constraint("noise", torchdist.constraints.greater_than(self.jitter))
+        self.set_constraint("noise", torchdist.constraints.positive)
```

Comment:
Constraint for noise is changed to `positive`. This is the only change for all GP algorithms.